src/states/elder/mod.rs

```diff
@@ -717,16 +717,22 @@ impl Elder {
         let dst = msg.message_dst();
         let signed_msg = msg.signed_routing_message();
         if self.in_location(dst) {
-            self.check_signed_message_trust(signed_msg)?;
+            let status = self.check_signed_message_trust(signed_msg)?;
             self.check_signed_message_integrity(signed_msg)?;
```

Comment:
what if the status return in above step is `Ok(TrustStatus::Unknown)`, but here failing the integrity check?

Reply:
Then we would not handle the message which is the correct thing to do.